dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1106,38 +1106,56 @@ private void provisionWorkflowVersionUploadFiles(ApiClient zendoClient, Deposit
         }
 
         // Add workflow version source files as a zip to the DOI upload deposit
-        // Borrow code from getWorkflowZip
         Set<SourceFile> sourceFiles = workflowVersion.getSourceFiles();
-        java.nio.file.Path path = Paths.get(workflowVersion.getWorkingDirectory());
         if (sourceFiles == null || sourceFiles.size() == 0) {
             LOG.warn("No source files found to zip when creating DOI");
             throw new CustomWebApplicationException("No source files found to"
                     + " upload when creating DOI. Zenodo requires at lease one file"
                     + " to be uploaded in order to create a DOI.", HttpStatus.SC_BAD_REQUEST);
         } else {
-            String fileName = workflow.getWorkflowPath().replaceAll("/", "-") + ".zip";
             OutputStream outputStream;
+            String versionOfWorkflow = workflowVersion.getName();
+            String filePrefix = workflow.getWorkflowPath().replaceAll("/", "-")
```

Comment:
The workflow files are put on Zenodo and associated with the upload file name. The Zenodo user can download the workflow files and the name of the file that is downloaded is that same name. Having an meaningful name helps the user know what they downloaded. So I will leave as is.

Reply:
Ah, I see, I didn't realize the name was significant to Zenodo.

That said,  you're specifying this as a prefix. Doesn't the generated filename have random characters in it? Can you give an example of a generated filename?

Reply:
github.com-DataBiosphere-topmed-workflows-UM_aligner_wdl_1.29.0_81941263699702983.zip

Reply:
>github.com-DataBiosphere-topmed-workflows-UM_aligner_wdl_1.29.0_81941263699702983.zip<

Is the value`81941263699702983` random characters or a timestamp? I suspect random characters. If they are random characters and the name is important in Zenodo, then I think the proper way to do this is to call `Files.createTempDirectory` instead of `File.createTempFile`, then create the name exactly the way you want it in the temp directory. Then just delete the temp directory afterwards.

I realize it's probably another day's work and we're coming up against deadlines, so deferring to @denis-yuen on whether you should do this now, later, or at all.

Reply:
Hmmm, for example, the filename `dockstore-1.6.0.zip` is visible at https://zenodo.org/record/2630727

It might be nice to have the file visible be `github.com-DataBiosphere-topmed-workflows-UM_aligner_wdl_1.29.0.zip` and you'd probably do that by the procedure above. It's probably worth doing.

Reply:
That said, I also see `ga4gh-dockstore-5da12ba ` in the file-tree :man_shrugging:

Reply:
Fixed the code so a temporary directory is created and a file name with no random characters is created and both are deleted after upload